File: Adm/Models/src/CRUD.java
package src;

import java.util.ArrayList;
import java.util.List;

public abstract class CRUD<T> {
    protected List<T> objetos = new ArrayList<>();

    public void inserir(T obj) {
        abrir();
        int id = 0;
        for (T x : objetos) {
            if (x.getId() > id) {
                id = x.getId();
            }
        }
        obj.setId(id + 1);
        objetos.add(obj);
        salvar();
    }

    public List<T> listar() {
        abrir();
        return objetos;
    }

    public T listarId(int id) {
        abrir();
        for (T x : objetos) {
            if (x.getId() == id) {
                return x;
            }
        }
        return null;
    }

    public void atualizar(T obj) {
        T x = listarId(obj.getId());
        if (x != null) {
            objetos.remove(x);
            objetos.add(obj);
            salvar();
        }
    }

    public void excluir(T obj) {
        T x = listarId(obj.getId());
        if (x != null) {
            objetos.remove(x);
            salvar();
        }
    }

    public abstract void salvar();
    public abstract void abrir();
}

File: Adm/Models/CRUD.cs
using System;
using System.Collections.Generic;

namespace Adm.Models
{
    public abstract class CRUD<T> where T : Inter
    {
         protected List<T> objetos = new List<T>();

        public void Inserir(T obj)
        {
            Abrir();
            int id = 0;
            foreach (T x in objetos)
            {
                if (x.getId() > id)
                {
                    id = x.getId();
                }
            }
            obj.setId(id + 1);
            objetos.Add(obj);
            Salvar();

            
        }

        public List<T> Listar()
        {
            Abrir();
            return objetos;
        }

        public T ListarId(int id)
        {
            Abrir();
            foreach (T x in objetos)
            {
                if (x.getId() == id)
                {
                    return x;
                }
            }
            return default(T);
        }

        public void Atualizar(T obj)
        {
            T x = ListarId(obj.getId());
            if (x != null)
            {
                objetos.Remove(x);
                objetos.Add(obj);
                Salvar();
            }
        }

        public void Excluir(T obj)
        {
            T x = ListarId(obj.getId());
            if (x != null)
            {
                objetos.Remove(x);
                Salvar();
            }
        }

        public abstract void Salvar();
        public abstract void Abrir();
    }
}
